bot/seasons/easter/bunny_name_generator.py

```diff
@@ -0,0 +1,106 @@
+import json
+import logging
+import random
+import re
+from pathlib import Path
+
+from discord.ext import commands
+
+log = logging.getLogger(__name__)
+
+with open(Path("bot", "resources", "easter", "bunny_names.json"), "r", encoding="utf8") as f:
+    BUNNY_NAMES = json.load(f)
+
+
+class BunnyNameGenerator(commands.Cog):
+    """Generate a random bunny name, or bunnify your Discord username!"""
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    def find_separators(self, displayname):
+        """Check if Discord name contains spaces so we can bunnify an individual word in the name."""
+
+        new_name = re.split(r'[_.\s]', str(displayname))
+        if displayname not in new_name:
+            return new_name
+
+    def find_vowels(self, displayname):
+        """
+        Finds vowels in the user's display name.
+
+        If the Discord name contains a vowel and the letter y,
+        it will match one or more of these patterns.
+        Only the most recently matched pattern will apply the changes.
+        """
+
+        new_name = None
+
+        option1 = re.sub(r'a.+y', 'patchy', displayname)
```

Comment:
Not a huge fan of the hard-coded and duplicate logic in `find_vowels`. The substitution logic should be put into a `for` loop that iterates over, say, a list of tuples with vowel/sub pairs. Other than that, this looks great!

Reply:
Something like this should maintain the desired behavior and looks a bit cleaner.

```py
expressions = [
    (r'a.+y', 'patchy'),
    (r'e.+y', 'ears'),
    (r'i.+y', 'ditsy'),
    (r'o.+y', 'oofy'),
    (r'u.+y', 'uffy'),
]

for exp, vowel_sub in expressions:
    new_name = re.sub(exp, vowel_sub, displayname)
    if new_name != displayname:
        return new_name
```